src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlingTests.Deserialize.cs

```diff
@@ -0,0 +1,1264 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class ReferenceHandlingTests
```

Comment:
Where possible, please combine similar read/write tests on `.Serialize` and `.Deserialize` into single tests that show roundtripping. I think the serialize/deserialize split should be used to highlight special cases (where roundtripping is not attainable or some peculiar one-sided behavior). See https://github.com/dotnet/corefx/pull/41977#discussion_r337745064

Reply:
I just hit a couple of bugs while switching to round-trip scenarios. Good call.